Specify charset in lighttpd mimetypes

mattiaverga opened the pull request:
Specify charset in lighttpd mimetypes for build log files and .spec files.
Fixes #3077

<!-- issue-commentator = {"comment-id":"2921406389"} -->

mattiaverga commented:
I have tested this by copying a builder-live.log.gz with bad characters from the real COPR into the copr_backend_httpd local docker and it works.

praiskup commented:
The file being modified is not used @ Fedora Copr, we need to better take a look at the problem first.

praiskup commented:
> It does seem to be used in Fedora deployment, at least:

Yeah, it is a copy of this documentation file.  It is good idea to fix it here, too ... but ideally we should ship this file as a configuration file, and start using it in our deployments :-)

> Perhaps I should open a PR to change the settings there?

Yes!  But ... I re-uploaded the affected log file on our staging, and tried to configure mime according to your advice, but it doesn't seem to help:
https://copr-be-dev.cloud.fedoraproject.org/experiments/06771727-builder-live.log.gz

praiskup commented:
I did this: https://pagure.io/fedora-infra/ansible/c/0ada9a2fb368e215c10cef7fba8decc5ddb0ceb4?branch=main

Would you mind updating this PR so we have the doc fixed the same way?

praiskup commented:
@mattiaverga thank you very much for finding the solution!

mattiaverga commented:
I've updated the PR as requested (I did not noticed that .spec already had a line).
However, I'm unsure about the .log.gz line. Currently I've removed it. Should we maintain that line too in mime.conf? Or should I add the setting in lighttpd.conf as you did in ansible?

praiskup commented:
The lighttpd conf versions diverged a lot between here and ansible.git, it is up to you how far you want to go with updating the local variant. The current variant looks good.
